alg/contour.cpp

```diff
@@ -726,40 +730,46 @@ CPLErr GDALContourGenerateEx(GDALRasterBandH hBand, void *hLayer,
                                            FixedLevelRangeIterator>
                     cg(hBand, useNoData, noDataValue, writer, levels);
                 ok = cg.process(pfnProgress, pProgressArg);
+                flOk = ok;
             }
-            else if (expBase > 0.0)
-            {
-                // Do not provide the actual minimum value to level iterator
-                // in polygonal case, otherwise it can result in a polygon
-                // with a degenerate min=max range.
-                ExponentialLevelRangeIterator levels(
-                    expBase, -std::numeric_limits<double>::infinity());
-                SegmentMerger<RingAppender, ExponentialLevelRangeIterator>
-                    writer(appender, levels, /* polygonize */ true);
-                ContourGeneratorFromRaster<decltype(writer),
-                                           ExponentialLevelRangeIterator>
-                    cg(hBand, useNoData, noDataValue, writer, levels);
-                ok = cg.process(pfnProgress, pProgressArg);
-            }
-            else
+
+            if (flOk)
```

Comment:
hum, so you are doing a 2 pass approach: first user provided fixed levels, and then contour intervals.
I'm not completely sure this will lead to correct results. One obvious (non nominal case) is that the same level is in fixed level and in contour intervals, which will lead to duplicated lines. But even in the nominal case, when using with -p (polygonization), the fact that the levels are not sorted might lead to wrong results.
This should be checked.
My initial idea for the implementation would have rather to modify the fixed level array to merge into it the levels generated from the contour interval, and sort it.